modules/trusted-firmware-m/Kconfig

```diff
@@ -305,4 +305,13 @@ config TFM_FLASH_MERGED_BINARY
 		(if building with TFM_BL2 is enabled), and the
 		Non-Secure application firmware.
 
+config TFM_EXTRA_CMAKE_ARGS
+	string
```

Comment:
This is not an optimal solution.
While it does solve some problems, it also carries some side-effects.

Namely that if moduleA needs to pass `-DFOO=bar` and thus creates:
```
# Kconfig.moduleA
config TFM_EXTRA_CMAKE_ARGS
    string
    default "-DFOO=bar"
```
and moduleB needs to pass `-DBAR=foo` with:
```
# Kconfig.moduleA
config TFM_EXTRA_CMAKE_ARGS
    string
    default "-DBAR=foo"
```

then only one of the values will be passed to TF-M.
Now, as long as either module A or module B is used, things will be fine, but if both are in use, this will break.

So this must be used with care.

I would prefer a cleaner and scalable solution, but could require some rework on how ExternalProject is used, cause the problem in the external project is how it handles generator expressions containing lists.

Reply:
@tejlmand I'm getting your point. Though, I think even with the limitation you mention, this feature is better than having users trying to hack `modules/trusted-firmware-m/CMakeLists.txt` to pass their custom TFM arguments (what I did with varying results)
Isn't possible that possible to state that for now, this should be defined only in one place ?

Reply:
I had the same argument, actually. We can keep working in the mean time to make this better. Otherwise we just keep defining explicit Kconfig options for the corresponding cmake arguments we want to pass

Reply:
Please note also that, in case of the "user hacks CMakeLists.txt" case, trying to identify if the arguments are correctly passed to TFM is not the easiest thing.
So having a validated method, even with limitation, is really a thing we miss.

Reply:
@tejlmand would an `EXTRA_TFM_CMAKE_ARGS` cmake list work for what you had in mind.

Reply:
@tejlmand, done, please take a look.

Reply:
Splitting between a "master" config and a "vendor" config has been discussed by me and @tejlmand offline... It helps a little, but it doesn't "solve it"

Reply:
Changed to use a cmake list instead, so adding from multiple places should now work fine.

    set_property(GLOBAL APPEND PROPERTY EXTRA_TFM_CMAKE_ARGS -DFOO=bar)

Reply:
@oyvindronningstad  problem with this approach is that it is introducing Zephyr module cmake order processing dependencies.

Reply:
I understand that this is undesirable